trl/trainer/ppo_trainer.py

```diff
@@ -590,7 +590,8 @@ def step(
         bs = self.config.batch_size
 
         queries, responses, scores = self._step_safety_checker(bs, queries, responses, scores)
-        scores = torch.tensor(scores)
+        scores_device = scores[0].device
```

Comment:
It would be safer to not access the first entry in the list and just use `self.current_device`

Reply:
that sounds like a good idea!

Reply:
I agree the fix is much better. Can you suggest a change to the PR, I can add it then.

Reply:
Thanks for fixing this issue!